modules/phase_field/src/postprocessors/GrainTracker.C

```diff
@@ -485,14 +470,12 @@ GrainTracker::remapGrains()
   do
   {
     grains_remapped = false;
-    for (std::map<unsigned int, MooseSharedPointer<FeatureData> >::iterator grain_it1 = _unique_grains.begin();
-         grain_it1 != _unique_grains.end(); ++grain_it1)
+    for (auto grain_it1 = _unique_grains.begin(); grain_it1 != _unique_grains.end(); ++grain_it1)
     {
       if (grain_it1->second->_status == INACTIVE)
         continue;
 
-      for (std::map<unsigned int, MooseSharedPointer<FeatureData> >::iterator grain_it2 = _unique_grains.begin();
-           grain_it2 != _unique_grains.end(); ++grain_it2)
+      for (auto grain_it2 = _unique_grains.begin(); grain_it2 != _unique_grains.end(); ++grain_it2)
       {
         // Don't compare a grain with itself and don't try to remap inactive grains
         if (grain_it1 == grain_it2 || grain_it2->second->_status == INACTIVE)
```

Comment:
Yeah - neither is better than the other.  This is a good use of the iterator case.  Iterators are still very much a part of the language... and we should use them when it makes sense.